store/builder/storage_manager_builder.go

```diff
@@ -206,19 +210,52 @@ func buildEigenDAV2Backend(
 		return nil, fmt.Errorf("build eth client: %w", err)
 	}
 
-	certVerifierAddressProvider := verification.NewStaticCertVerifierAddressProvider(
-		geth_common.HexToAddress(config.ClientConfigV2.EigenDACertVerifierAddress))
+	routerOrImmutableVerifierAddr := geth_common.HexToAddress(config.ClientConfigV2.EigenDACertVerifierRouterAddress)
+	caller, err := binding.NewContractEigenDACertVerifierRouterCaller(routerOrImmutableVerifierAddr, ethClient)
+	if err != nil {
+		return nil, fmt.Errorf("new cert verifier router caller: %w", err)
+	}
 
-	certVerifier, err := verification.NewCertVerifier(
-		log, ethClient, certVerifierAddressProvider)
+	isRouter := true
+	// Check if the router address is actually a router. if method `getRouterAddress` fails, it means that the
+	// address is not a router, and we should treat it as an immutable cert verifier instead
+	_, err = caller.GetCertVerifierAt(&bind.CallOpts{Context: ctx}, 0)
 	if err != nil {
-		return nil, fmt.Errorf("new cert verifier: %w", err)
+		log.Warn(`EigenDA cert verifier router address was detected to not be a router, using it as an
+			immutable cert verifier instead`)
```

Comment:
don't think this works. This call could fail for many reasons unrelated to this right? like a network error, or a context timeout, etc. Think we need to be a lot more strict and make sure that the error is `function doesn't exist` (not sure if that error exists though...)

Reply:
hmm https://pkg.go.dev/github.com/ethereum/go-ethereum/accounts/abi/bind#pkg-variables doesn't seem to have an error for this.. but there must be a way to figure this out?

Reply:
🤔

Reply:
good callout - we could leverage what arbitrum uses for their BoLD activation check:
https://github.com/OffchainLabs/nitro/blob/master/staker/multi_protocol/multi_protocol_staker.go#L171-L188

Reply:
but even this makes assumptions on the provider - in the past we've seen issues where customers inject networking middleware between proxy and RPC.... 

https://github.com/OffchainLabs/nitro/blob/master/util/headerreader/header_reader.go#L29-L43

Reply:
Maybe an easy solution for now is just to separate the CertVerifier and CertRouter flags and let the user input the correct choice?